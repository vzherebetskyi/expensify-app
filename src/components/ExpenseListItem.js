import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({ description, amount, createAt, id }) => (
    <div>
        {/* { expenses.map(element => 
            <p key = {element.description}>{`${element.description}, ${element.amount}, ${element.createAt}`}</p>
            )
        } */}
        <h3><Link to = {`/edit/${id}`}>{description}</Link></h3>
        <p>
          {numeral(amount / 100).format('$0,0.00')} 
          - 
          {moment(createAt).format('MMMM Do YYYY')}
        </p>
    </div>
);

export default ExpenseListItem;