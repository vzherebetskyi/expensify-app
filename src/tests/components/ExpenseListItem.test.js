import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import { expenses } from '../test_cases';

test('should render an ExpenseListItem', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0] } />);
    expect(wrapper).toMatchSnapshot();
});