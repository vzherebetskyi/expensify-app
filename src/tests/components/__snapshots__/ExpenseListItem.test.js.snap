// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an ExpenseListItem 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/1"
    >
      Food
    </Link>
  </h3>
  <p>
    $2.00
    -
    January 1st 1970
  </p>
</div>
`;
